### PR TITLE
feat/print-stout-files (First PR)

### PR DESCRIPTION
**Pull request by FeliceGeracitano**
TODO
- [x] sort by bigger delta
- [x] move delta on current branch

### CODE REVIEW

File: src/lib/utils.ts [hunk 1]
```diff
@@ -48,7 +48,7 @@ export const getFormattedRows = (
       withDeltaSize(report.targetBranchReport[fileName], report.currentBranchReport[fileName])
     ])
     .map(([fileName, targetBranchSize, currentBranchSize]) => [
-      fileName.replace(omitFromFilename, ''),
+      fileName.slice((fileName.match(stampFromMatch) || ({} as any)).index),
```
Comment (FeliceGeracitano) [inline src/lib/utils.ts]: we don't have `workdir` from the cli, so we need to match the dist from the full path and remove what before that

File: src/lib/utils.ts [hunk 2]
```diff
@@ -38,7 +38,7 @@ export const groupFilesByExtension = (targetedFiles: string[]): { [key: string]:
 
 export const getFormattedRows = (
   report: IBundleCheckerReport,
-  omitFromFilename: string = ''
+  stampFromMatch: string = ''
```
Comment (rbelling) [inline src/lib/utils.ts]: this won't be needed if #59 is merged

File: src/lib/utils.ts (hunk 1 above)
Comment (rbelling) [reply src/lib/utils.ts]: this won't be needed if #59 is merged

**Comment by rbelling** [comment]
| File name | master | feat/print-stout-files |
| --- | --- | --- |
| js | 12.81KB (🔺 +727B) | 12.1KB |

**Comment by rbelling** [comment]
| File extension | master | feat/print-stout-files |
| --- | --- | --- |
| /build/lib/utils.js | 4.33KB (🔺 +681B) | 3.67KB |
| /build/commands/compare.js | 3.36KB (🔺 +32B) | 3.33KB |
| /build/lib/index.js | 4.97KB (🔺 +14B) | 4.96KB |

**Comment by rbelling** [comment]
| File name | master | feat/print-stout-files |
| --- | --- | --- |
| js | 12.69KB (🔺 +602B) | 12.1KB |

**Comment by rbelling** [comment]
| File extension | master | feat/print-stout-files |
| --- | --- | --- |
| /build/lib/utils.js | 4.24KB (🔺 +588B) | 3.67KB |
| /build/lib/index.js | 4.97KB (🔺 +14B) | 4.96KB |

**Comment by rbelling** [comment]
| File name | master | feat/print-stout-files |
| --- | --- | --- |
| js | 12.69KB (🔺 +602B) | 12.1KB |

**Comment by rbelling** [comment]
| File extension | master | feat/print-stout-files |
| --- | --- | --- |
| /build/lib/utils.js | 4.24KB (🔺 +588B) | 3.67KB |
| /build/lib/index.js | 4.97KB (🔺 +14B) | 4.96KB |

**Comment by rbelling** [comment]
| File name | feat/print-stout-files | master |
| --- | --- | --- |
| js | 12.69KB (🔺 +602B) | 12.1KB |

**Comment by rbelling** [comment]
| File extension | feat/print-stout-files | master |
| --- | --- | --- |
| /build/lib/utils.js | 4.24KB (🔺 +588B) | 3.67KB |
| /build/lib/index.js | 4.97KB (🔺 +14B) | 4.96KB |

**Comment by rbelling** [comment]
| File name | feat/print-stout-files | master |
| --- | --- | --- |
| js | 12.69KB (🔺 +602B) | 12.1KB |

**Comment by rbelling** [comment]
| File extension | feat/print-stout-files | master |
| --- | --- | --- |
| /build/lib/utils.js | 4.24KB (🔺 +588B) | 3.67KB |
| /build/lib/index.js | 4.97KB (🔺 +14B) | 4.96KB |

File: src/commands/compare.ts [hunk 3]
```diff
@@ -38,8 +38,8 @@ export default class Compare extends Command {
     const { currentBranch, targetBranch } = localFlags;
     const checker = new BundleChecker(localFlags);
     const report = await checker.compare();
-    const overviewReportHeader = ['File name', targetBranch, currentBranch] as ITableRow;
-    const filesBreakDownHeader = ['File extension', targetBranch, currentBranch] as ITableRow;
+    const overviewReportHeader = ['File name', currentBranch, targetBranch] as ITableRow;
+    const filesBreakDownHeader = ['File extension', currentBranch, targetBranch] as ITableRow;
```
Comment (rbelling) [inline src/commands/compare.ts]: targetBranch is `master` - are we moving it to the third column?

File: src/lib/utils.ts [hunk 4]
```diff
@@ -94,19 +100,34 @@ const getConsoleTotalTable = ({
...
+  if (bDelta - aDelta > 0) return 1;
+  if (bDelta - aDelta < 0) return -1;
+  return 0;
+};
+
+const filterDelta = (itemRow: Array<string | number>) => {
+  const delta = (itemRow[1] as number) - (itemRow[2] as number);
+  if (delta === 0) return false;
+  return true;
+};
```
Comment (rbelling) [inline src/lib/utils.ts]: can we move those two `filterDelta` and `sortByDelta` to the top of the file?

File: src/lib/utils.ts [hunk 5]
```diff
@@ -94,19 +100,34 @@ const getConsoleTotalTable = ({
...
+  report
+}: IPrintStdout): any =>
+  getFormattedRows(report).map(([fileName, currentSize, targetSize]) => ({
+    File: fileName,
+    [currentBranchName]: currentSize,
+    [targetBranchName]: targetSize
+  }));
+
+// TODO: fix types, and make them more robust
+const sortByDelta = (a: Array<string | number>, b: Array<string | number>) => {
```
Comment (rbelling) [inline src/lib/utils.ts]: can the type of `a` and `b` be updated to `ITableRow` ?

File: src/lib/utils.ts [hunk 6]
```diff
@@ -94,19 +100,34 @@ const getConsoleTotalTable = ({
...
+  const bDelta = (b[1] as number) - (b[2] as number);
+  const aDelta = (a[1] as number) - (a[2] as number);
+  if (bDelta - aDelta > 0) return 1;
+  if (bDelta - aDelta < 0) return -1;
+  return 0;
+};
+
+const filterDelta = (itemRow: Array<string | number>) => {
```
Comment (rbelling) [inline src/lib/utils.ts]: how about `ITableRow` for the type?

File: src/lib/utils.ts (hunk 5 above)
Comment (FeliceGeracitano) [reply src/lib/utils.ts]: Unfortunately thet are not ITableRow, ITableRow is: 
```
ITableRow = [string, string, string]
```
while here we are dealing with:
```
 [string, number, number]
```
to allow delta calculation
this is why i left a TODO to eventually define those type in between the pipe trasformation

File: src/lib/utils.ts (hunk 6 above)
Comment (FeliceGeracitano) [reply src/lib/utils.ts]: same at above

File: src/commands/compare.ts (hunk 3 above)
Comment (FeliceGeracitano) [reply src/commands/compare.ts]: I prefer to have the `current branch` in 2nd columns since the delta is associated with his changes

**Comment by rbelling** [comment]
| File name | feat/print-stout-files | master |
| --- | --- | --- |
| js | 12.69KB (🔺 +602B) | 12.1KB |

**Comment by rbelling** [comment]
| File extension | feat/print-stout-files | master |
| --- | --- | --- |
| /build/lib/utils.js | 4.24KB (🔺 +588B) | 3.67KB |
| /build/lib/index.js | 4.97KB (🔺 +14B) | 4.96KB |

**Comment by rbelling** [comment]
| File name | feat/print-stout-files | master |
| --- | --- | --- |
| js | 12.57KB (🔺 +482B) | 12.1KB |

**Comment by rbelling** [comment]
| File extension | feat/print-stout-files | master |
| --- | --- | --- |
| /build/lib/utils.js | 4.12KB (🔺 +468B) | 3.67KB |
| /build/lib/index.js | 4.97KB (🔺 +14B) | 4.96KB |

**Comment by rbelling** [comment]
| File name | feat/print-stout-files | master |
| --- | --- | --- |
| js | 12.57KB (🔺 +482B) | 12.1KB |

**Comment by rbelling** [comment]
| File extension | feat/print-stout-files | master |
| --- | --- | --- |
| /build/lib/utils.js | 4.12KB (🔺 +468B) | 3.67KB |
| /build/lib/index.js | 4.97KB (🔺 +14B) | 4.96KB |

**Comment by rbelling** [comment]
| File name | feat/print-stout-files | master |
| --- | --- | --- |
| js | 12.58KB (🔺 +492B) | 12.1KB |

**Comment by rbelling** [comment]
| File extension | feat/print-stout-files | master |
| --- | --- | --- |
| /build/lib/utils.js | 4.13KB (🔺 +478B) | 3.67KB |
| /build/lib/index.js | 4.97KB (🔺 +14B) | 4.96KB |

**Comment by rbelling** [comment]
| File name | feat/print-stout-files | master |
| --- | --- | --- |
| js | 12.58KB (🔺 +492B) | 12.1KB |

**Comment by rbelling** [comment]
| File extension | feat/print-stout-files | master |
| --- | --- | --- |
| /build/lib/utils.js | 4.13KB (🔺 +478B) | 3.67KB |
| /build/lib/index.js | 4.97KB (🔺 +14B) | 4.96KB |

**Pull request merged**